stdlib/Printf/src/Printf.jl

```diff
@@ -726,6 +791,14 @@ const UNROLL_UPTO = 16
     return pos
 end
 
+@inline function plength(f::Spec{T}, args, argp) where {T}
+    if f.dynamic_width || f.dynamic_precision
```

Comment:
This `if` statement is redundant.  You can just do
```jl
f, argp = rmdynamic(f, args, argp)
```
unconditionally.

Reply:
yes,  but the we will call rmdynamic()  even if they are both false.

This will work ok but don't we then also construct a new Spec data structure in rmdynamic()? How smart is the compiler?

Reply:
Constructing a new `Spec` data structure should be essentially free since it is an immutable bitstype struct.

Reply:
(It wouldn't hurt to check this with a benchmark, of course.)

Reply:
You got a point, didn't realize that it was a bitstype - I'll do some benchmarks.

Reply:
Did some benchmarks and the difference is not measurable. In the end, since a printf is likely to end up as a I/O-operation the minute performance difference (if any)  will any way drown in the time it takes to do the I/O operation.  I removed it to reduce the code.